src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -640,60 +640,80 @@ def _set_linux_user(
     if ssh_key_value:
         protected_settings['ssh_key'] = read_content_if_is_file(ssh_key_value)
 
-    poller = _update_linux_access_extension(resource_group_name, vm_name,
+    poller = _update_linux_access_extension(vm_instance, resource_group_name,
                                             protected_settings)
     return ExtensionUpdateLongRunningOperation('setting user', 'done')(poller)
 
 
-def _reset_windows_admin(
-        resource_group_name, vm, username, password):
+def _reset_windows_admin(vm_instance, resource_group_name, username, password):
```

Comment:
So was the issue that this is a patch-like update where you have to get the instance first and then mutate it?